hydra/hydra.go

```diff
@@ -0,0 +1,153 @@
+package hydra
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/axiomhq/hyperloglog"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	levelds "github.com/ipfs/go-ds-leveldb"
+	"github.com/libp2p/go-libp2p-core/network"
+	kbucket "github.com/libp2p/go-libp2p-kbucket"
+	"github.com/libp2p/hydra-booster/metrics"
+	"github.com/libp2p/hydra-booster/sybil"
+	hysybilopts "github.com/libp2p/hydra-booster/sybil/opts"
+	"github.com/multiformats/go-multiaddr"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+type Hydra struct {
+	Sybils       []*sybil.Sybil
+	Datastore    datastore.Datastore
```

Comment:
```suggestion
	SharedDatastore    datastore.Datastore
```

Suggestion, rename to SharedDatastore so that it is clear that at this level, we are seeing the datastore that is shared by all (vs. the datastore per sybil, which might be this one or something else)

Reply:
Done in https://github.com/libp2p/hydra-booster/pull/35